### PR TITLE
ADAPT-2851: Add canonical url to SEO

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# NOT READY FOR REVIEW

# Summary
- Add canonical url to SEO

# Review By (Date)
- July 13

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. [Netlify Test Preview](https://deploy-preview-71--stanford-alumni.netlify.app/test/rebecca/url-test)

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- [ADAPT-2851](https://stanfordits.atlassian.net/browse/ADAPT-2851)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/partials/seo.js [hunk 1]
```diff
@@ -42,9 +71,13 @@ const Seo = ({ blok: { title: theTitle, seo }, blok }) => {
     twitterImage = transformImage(twitterImage, "/1200x600");
   }
 
+  const canonicalUrl = getCanonicalUrl(blok, siteUrl, location);
```
Comment (sherakama) [inline src/components/partials/seo.js:74]: This is rendering: https://alumni-preview.stanford.eduundefined for pages without the URL set. 

* https://deploy-preview-71--stanford-alumni.netlify.app/test/yvonne/test-basic-page/

**Review by sherakama — CHANGES_REQUESTED** [review]
Not working as expected yet. Please revise.

**Comment by yvonnetangsu** [comment]
@rebeccahongsf Thanks for adding the schema to the basic page and dark background page. Please add it to the Story content type too 🙏🏼 😄

**Review by sherakama — CHANGES_REQUESTED** [review]
This is working well

* https://deploy-preview-71--stanford-alumni.netlify.app/test-with-canonical/
* https://deploy-preview-71--stanford-alumni.netlify.app/test/rebecca/url-test/
* https://deploy-preview-71--stanford-alumni.netlify.app/about/people/departments 
* https://deploy-preview-71--stanford-alumni.netlify.app/sitemap/sitemap-0.xml

Please revise the linting errors and then this is GTG.

**Review by yvonnetangsu — APPROVED** [review]
Approving since @sherakama said this is GTG pending eslint warnings which you have now fixed - thank you!

**Review by sherakama — APPROVED** [review]
GTG

**Pull request merged**